src/main/java/net/tirasa/ilgrosso/resetdb/Main.java

```diff
@@ -39,158 +39,159 @@ public class Main {
+                    + "FROM information_schema.views;")) {
+                while (resultSet.next()) {
+                    drops.add(resultSet.getString(1));
+                }
+            }
 
-        for (String drop : drops) {
-            statement.executeUpdate(drop.substring(0, drop.length() - 1));
-        }
-        drops.clear();
-
-        drops.add("SET FOREIGN_KEY_CHECKS = 0;");
-        resultSet = statement.executeQuery(
-                "SELECT concat('DROP TABLE IF EXISTS ', table_name, ' CASCADE;')"
-                + "FROM information_schema.tables;");
-        while (resultSet.next()) {
-            drops.add(resultSet.getString(1));
-        }
-        resultSet.close();
-        drops.add("SET FOREIGN_KEY_CHECKS = 1;");
+            for (String drop : drops) {
+                statement.executeUpdate(drop.substring(0, drop.length() - 1));
+            }
+            drops.clear();
+
+            drops.add("SET FOREIGN_KEY_CHECKS = 0;");
+
+            try ( ResultSet resultSet = statement.executeQuery(
+                    "SELECT concat('DROP TABLE IF EXISTS ', table_name, ' CASCADE;')"
+                    + "FROM information_schema.tables;")) {
+                while (resultSet.next()) {
+                    drops.add(resultSet.getString(1));
+                }
+            }
+            drops.add("SET FOREIGN_KEY_CHECKS = 1;");
+
+            for (String drop : drops) {
+                statement.executeUpdate(drop.substring(0, drop.length() - 1));
+            }
 
-        for (String drop : drops) {
-            statement.executeUpdate(drop.substring(0, drop.length() - 1));
         }
 
-        statement.close();
-        conn.close();
     }
 
     private static void resetOracle(final Connection conn)
             throws Exception {
 
-        final Statement statement = conn.createStatement();
-
-        ResultSet resultSet = statement.executeQuery(
-                "SELECT 'DROP VIEW ' || object_name || ';'"
-                + " FROM user_objects WHERE object_type='VIEW'");
-        final List<String> drops = new ArrayList<>();
-        while (resultSet.next()) {
-            drops.add(resultSet.getString(1));
-        }
-        resultSet.close();
+        try (final Statement statement = conn.createStatement()) {
 
-        for (String drop : drops) {
-            statement.executeUpdate(drop.substring(0, drop.length() - 1));
-        }
-        drops.clear();
-
-        resultSet = statement.executeQuery(
-                "SELECT 'DROP INDEX ' || object_name || ';'"
-                + " FROM user_objects WHERE object_type='INDEX'"
-                + " AND object_name NOT LIKE 'SYS_%'");
-        while (resultSet.next()) {
-            drops.add(resultSet.getString(1));
-        }
-        resultSet.close();
+            final List<String> drops = new ArrayList<>();
+            try ( ResultSet resultSet = statement.executeQuery(
```

Comment:
`try ( ` (e.g. with a space after parenthesis) is going to create troubles with checkstyle - I know it is not currently set up o this project, anyway...

Reply:
Ciao

I can 

- remove space after parenthesis (only on this PR)
- create another PR on master, add checkstyle and fill all errors (parentheses and javadoc)

tell me what I can do 

ciao
matteo

Reply:
Please go on with option (1), thanks.
Adding checkstyle can be handled separately.